### PR TITLE
Remove jquery

### PR DESCRIPTION
**Pull request by mdartic**
This PR is related to #86 

I remove jQuery dependency and change the legacy code into the standard API.

$.get => xhr
$(this).find => querySelector / querySelectorAll
$(this).attr('name') => element.attributes.name

It's a WIP. 

Please review it first before merging, I can't guarantee all is working perfectly.

I've added a small test for `parseAndExplodeTimeRange` method, to only affect duration if it's not empty (it was breaking on my test case).

I comment the old code, and if you agree with this PR, I will delete these comments.

Thanks for your review

### CODE REVIEW

File: dist/leaflet.timedimension.src.js [hunk 1]
```diff
@@ -454,7 +454,7 @@ L.TimeDimension.Util = {
         var tr = timeRange.split('/');
         var startTime = new Date(Date.parse(tr[0]));
         var endTime = new Date(Date.parse(tr[1]));
-        var duration = tr.length > 2 ? tr[2] : "P1D";
+        var duration = tr.length > 2  && tr[2] !== "" ? tr[2] : "P1D";
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: Keep the pull request clean, this is unrelated to jQuery.

File: dist/leaflet.timedimension.src.js [hunk 2]
```diff
@@ -969,15 +969,29 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
             this._setDefaultTime = this._setDefaultTime || (this._timeDimension && this._timeDimension.getAvailableTimes().length == 0);
             this.setAvailableTimes(this._parseTimeDimensionFromCapabilities(data));
             if (this._setDefaultTime && this._timeDimension) {
                 this._timeDimension.setCurrentTime(this._defaultTime);
             }
         }).bind(this));
+        oReq.open("GET", url);
+        oReq.send();
+
+        // $.get(url, (function(data) {
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: Comments can be removed if they're no longer functional

File: dist/leaflet.timedimension.src.js [hunk 3]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
             times = this._getTimesFromLayerCapabilities(layer);
             if (!times) {
-                times = this._getTimesFromLayerCapabilities(layer.parent());
+                times = this._getTimesFromLayerCapabilities(layerParent);
             }
         }
         return times;
+
+        // var layers = $(xml).find('Layer[queryable="1"]');
+        // var layerName = this._baseLayer.wmsParams.layers;
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: See previous comment on comments ;)

File: dist/leaflet.timedimension.src.js [hunk 4]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
-            var extent = layer.find("Extent[name='time']");
+            var extent = layer.querySelectorAll("Extent[name='time']");
             if (extent && extent.length && extent[0].textContent.length) {
                 times = extent[0].textContent.trim();
             }
         }
         return times;
+        // var times = null;
+        // var dimension = layer.querySelector("Dimension[name='time']");
+        // if (dimension && dimension.length && dimension[0].textContent.length) {
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: :)

File: dist/leaflet.timedimension.src.js [hunk 5]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
+                defaultTime = this._getDefaultTimeFromLayerCapabilities(layerParent);
             }
         }
         return defaultTime;
+
+        // var layers = $(xml).find('Layer[queryable="1"]');
+        // var layerName = this._baseLayer.wmsParams.layers;
+        // var layerNameElement = layers.find("Name").filter(function(index) {
+        //     return $(this).text() === layerName;
+        // });
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: :o

File: dist/leaflet.timedimension.src.js [hunk 6]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
+                defaultTime = extent[0].attributes.default.value;
             }
         }
         return defaultTime;
+
+        // var defaultTime = 0;
+        // var dimension = layer.find("Dimension[name='time']");
+        // if (dimension && dimension.attr("default")) {
+        //     defaultTime = dimension.attr("default");
+        // } else {
```
Comment (murb) [inline dist/leaflet.timedimension.src.js]: ...

File: dist/leaflet.timedimension.src.withlog.js [hunk 7]
```diff
@@ -969,15 +969,29 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
             if (this._setDefaultTime && this._timeDimension) {
                 this._timeDimension.setCurrentTime(this._defaultTime);
             }
         }).bind(this));
+        oReq.open("GET", url);
+        oReq.send();
+
+        // $.get(url, (function(data) {
+        //     this._defaultTime = Date.parse(this._getDefaultTimeFromCapabilities(data));
+        //     this._setDefaultTime = this._setDefaultTime || (this._timeDimension && this._timeDimension.getAvailableTimes().length == 0);
```
Comment (murb) [inline dist/leaflet.timedimension.src.withlog.js]: ...

File: dist/leaflet.timedimension.src.withlog.js [hunk 8]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
         return times;
+
+        // var layers = $(xml).find('Layer[queryable="1"]');
+        // var layerName = this._baseLayer.wmsParams.layers;
+        // var layerNameElement = layers.find("Name").filter(function(index) {
+        //     return $(this).text() === layerName;
+        // });
+        // var times = null;
+        // if (layerNameElement) {
+        //     var layer = layerNameElement.parent();
```
Comment (murb) [inline dist/leaflet.timedimension.src.withlog.js]: ...

File: dist/leaflet.timedimension.src.withlog.js [hunk 9]
```diff
@@ -992,65 +1006,127 @@ L.TimeDimension.Layer.WMS = L.TimeDimension.Layer.extend({
...
             if (extent && extent.length && extent[0].textContent.length) {
                 times = extent[0].textContent.trim();
             }
         }
         return times;
+        // var times = null;
+        // var dimension = layer.querySelector("Dimension[name='time']");
+        // if (dimension && dimension.length && dimension[0].textContent.length) {
+        //     times = dimension[0].textContent.trim();
+        // } else {
```
Comment (murb) [inline dist/leaflet.timedimension.src.withlog.js]: ...

File: src/leaflet.timedimension.util.js [hunk 10]
```diff
@@ -60,7 +60,7 @@ L.TimeDimension.Util = {
         var tr = timeRange.split('/');
         var startTime = new Date(Date.parse(tr[0]));
         var endTime = new Date(Date.parse(tr[1]));
-        var duration = tr.length > 2 ? tr[2] : "P1D";
+        var duration = tr.length > 2  && tr[2] !== "" ? tr[2] : "P1D";
```
Comment (murb) [inline src/leaflet.timedimension.util.js]: not related to jQuery

**Comment by murb** [comment]
... just removed the comment about not creating a pull request with commented-out code, as you already said so :)

**Comment by mdartic** [comment]
Well, sorry, I see only now you have reply to this PR... if you fork from my PR, I think I can close this PR ?

**Comment by murb** [comment]
No problem :) Jep this one can now be closed ;)

**Comment by murb** [comment]
btw: I didn't copy your `var duration = tr.length > 2  && tr[2] !== "" ? tr[2] : "P1D";` fix... which I thought didn't belong in a 'remove jquery' pull request :o

**Comment by mdartic** [comment]
yes, you're right.
so, for others, please check #119
